cli/root.go

```diff
@@ -1305,3 +1134,146 @@ func defaultUpgradeMessage(version string) string {
 	}
 	return fmt.Sprintf("download the server version with: 'curl -L https://coder.com/install.sh | sh -s -- --version %s'", version)
 }
+
+// wrapTransportWithEntitlementsCheck adds a middleware to the HTTP transport
+// that checks for entitlement warnings and prints them to the user.
+func wrapTransportWithEntitlementsCheck(rt http.RoundTripper, w io.Writer) http.RoundTripper {
+	var (
+		mu   sync.RWMutex
```

Comment:
I think these functions could be simplified with a plain mutex (vs rw). These code paths aren't really hot enough to benefit from read locking IMO.

Reply:
Fair I suppose this is just for client requests.